src/cmd/kubectl.rs

```diff
@@ -715,20 +715,31 @@ where
     }
 }
 
-pub fn kubectl_exec_get_pod<P>(
+pub fn kubectl_exec_get_pods<P>(
     kubernetes_config: P,
-    namespace: &str,
-    selector: &str,
+    namespace: Option<&str>,
+    selector: Option<&str>,
     envs: Vec<(&str, &str)>,
 ) -> Result<KubernetesList<KubernetesPod>, SimpleError>
 where
     P: AsRef<Path>,
 {
-    kubectl_exec::<P, KubernetesList<KubernetesPod>>(
-        vec!["get", "pod", "-o", "json", "-n", namespace, "-l", selector],
-        kubernetes_config,
-        envs,
-    )
+    let mut cmd_args = vec!["get", "pod", "-o", "json"];
```

Comment:
To have a list i think you need to set podS instead of pod

Reply:
Indeed :)